src/PopIII/popiii.cpp

```diff
@@ -0,0 +1,431 @@
+		amrex::Real const distxy = std::sqrt(std::pow(x - x0, 2) + std::pow(y - y0, 2));
+
+		burn_t state;
+		amrex::Real rhotot = 0.0;
+
+		for (int n = 0; n < NumSpec; ++n) {
+			state.xn[n] = numdens[n] * numdens_init;
+			rhotot += state.xn[n] * spmasses[n]; // spmasses contains the masses of all species, defined in EOS
+		}
+
+		// normalize -- just in case
+		std::array<Real, NumSpec> mfracs = {-1.0};
+		Real msum = 0.0;
+		for (int n = 0; n < NumSpec; ++n) {
+			mfracs[n] = state.xn[n] * spmasses[n] / rhotot;
+			msum += mfracs[n];
+		}
+
+		for (int n = 0; n < NumSpec; ++n) {
+			mfracs[n] /= msum;
+			// use the normalized mass fractions to obtain the corresponding number densities
+			state.xn[n] = mfracs[n] * rhotot / spmasses[n];
+		}
+
+		//amrex::Print() << "cell " << i << j << k << " " << rhotot << " " << numdens_init << " " << numdens[0] << std::endl;
+
+		amrex::Real phi = atan2((y-y0), (x-x0));
+
+		double vx = 0;
+		double vy = 0;
+		double vz = 0;
+
+		if (r <= R_sphere) {
+			state.rho = rhotot;
+			state.T = core_temp;
+			vx = renorm_amp * dvx(i, j, k);
+			vy = renorm_amp * dvy(i, j, k);
+			vz = renorm_amp * dvz(i, j, k);
+
+			// add rotation to vx and vy
+			vx += (-1.0) * distxy * omega_sphere * std::sin(phi);
+			vy += distxy * omega_sphere * std::cos(phi);
+
+		} else {
+			state.rho = 0.01 * rhotot;
+			state.T = 1e2 * core_temp; // can use EOS.hpp function to solve for temp
+		}
+
+		// call the EOS to set initial internal energy e
+		eos(eos_input_rt, state);
+		amrex::Real e = state.rho * state.e;
+
+		//amrex::Print() << "cell " << i << j << k << " " << state.rho << " " << state.T << " " << e << std::endl;
+
+		state_cc(i, j, k, HydroSystem<PopIII>::density_index) = state.rho;
+		state_cc(i, j, k, HydroSystem<PopIII>::x1Momentum_index) = state.rho * vx;
+		state_cc(i, j, k, HydroSystem<PopIII>::x2Momentum_index) = state.rho * vy;
+		state_cc(i, j, k, HydroSystem<PopIII>::x3Momentum_index) = state.rho * vz;
+		state_cc(i, j, k, HydroSystem<PopIII>::internalEnergy_index) = e;
+
+		Real const Egas = RadSystem<PopIII>::ComputeEgasFromEint(state.rho, state.rho*vx, state.rho*vy, state.rho*vz, e);
+		state_cc(i, j, k, HydroSystem<PopIII>::energy_index) = Egas;
+
+		for (int nn = 0; nn < NumSpec; ++nn) {
+			state_cc(i, j, k, HydroSystem<PopIII>::scalar0_index + nn) =
+			    mfracs[nn] * state.rho; // we use partial densities and not mass fractions
+		}
+
+	});
+}
+
+template <> void RadhydroSimulation<PopIII>::ErrorEst(int lev, amrex::TagBoxArray &tags, amrex::Real /*time*/, int /*ngrow*/)
+{
+	// refine on Jeans length
+	const int N_cells = 32; // inverse of the 'Jeans number' [Truelove et al. (1997)]
+	const amrex::Real dx = geom[lev].CellSizeArray()[0];
+	const amrex::Real G = Gconst_;
+
+	auto const &state = state_new_cc_[lev].const_arrays();
+	auto tag = tags.arrays();
+	amrex::Print() << "start error est " << std::endl;
```

Comment:
I don't think we want to commit print statement this unless there's a way to turn it on/off.

Reply:
ah lol, yes.

Reply:
will do it at the end before this PR is merged